lib/kitchen/provisioner.rb

```diff
@@ -34,7 +34,16 @@ module Provisioner
     # @return [Provisioner::Base] a provisioner instance
     # @raise [ClientError] if a provisioner instance could not be created
     def self.for_plugin(plugin, config)
+      plugin, config = change_provider_chef_zero(plugin, config)
       Kitchen::Plugin.load(self, plugin, config)
```

Comment:
Instead of a separate function, what about moving the `if` block of `change_provider_chef_zero` into this function? It's a little more concise, falls into the domain of what we'd expect this function to do (determining the right plugin to load) and can be tested by verifying the parameters of Plugin.load for a given set of inputs.

Reply:
@marcparadise it's updated now.